lib/submit-request.js

```diff
@@ -103,7 +146,7 @@ SubmitRequest.prototype.submit = function(callback) {
 
     // Transform the op up to the current snapshot version, then apply
     var from = op.v;
-    backend.db.getOpsToSnapshot(collection, id, from, snapshot, null, function(err, ops) {
+    backend.db.getOpsToSnapshot(collection, id, from, snapshot, {metadata: true}, function(err, ops) {
```

Comment:
I don't know if asking for metadata here has any wider-reaching consequences? We may also need to check support of this flag to determine whether we allow fixups? (I think it'll potentially break in a weird way if it's not supported)

Reply:
We should scrub this metadata before sending back to the client, which can happen eg if an unsubscribed doc is behind the server and submits an op.

Reply:
Alec to verify whether catch-up ops' metadata is sent down or not if a client that's behind in versions sends a new op